assets/dev/js/editor/controls/media.js

```diff
@@ -243,6 +251,54 @@ ControlMediaItemView = ControlMultipleBaseItemView.extend( {
 	onBeforeDestroy() {
 		this.$el.remove();
 	},
+
+	async onMediaImageSize() {
+		this.ui.mediaImageSize.css( 'color', 'inherit' );
+		const previewPlaceHolder = this.$el.find( '.elementor-control-media__preview' );
+		const selectedImageId = {
+			id: this?.getControlValue( 'id' ) || previewPlaceHolder[ 0 ].getAttribute( 'data-id' ),
+			alt: this?.getControlValue( 'alt' ) || previewPlaceHolder[ 0 ].getAttribute( 'alt' ),
+		};
+
+		const selectedSize = this.ui.mediaImageSize.val();
+		let imageURL = await elementor.imagesManager.getImageUrl( {
+			id: selectedImageId.id,
+			size: selectedSize,
+		} );
+		const stateOptions = {
+			url: null,
+			id: selectedImageId.id,
+			alt: selectedImageId.alt,
+			dimensions: selectedSize,
+			source: 'library',
+		};
+		if ( imageURL ) {
+			stateOptions.url = imageURL;
+			this.setValue( stateOptions );
+		} else {
+			elementor.channels.editor.once( 'imagesManager:detailsReceived', ( data ) => {
+				imageURL = data[ selectedImageId.id ][ selectedSize ];
+				if ( imageURL ) {
+					stateOptions.url = imageURL;
+					this.setValue( stateOptions );
+				}
+			} );
+		}
+	},
+
+	toggleSizeControl() {
+		const previewImageExist = this.$el.find( '.elementor-control-media__preview' )[ 0 ]?.style?.backgroundImage;
+		const backgroundControl = this.$el[ 0 ]?.className?.match( /background_image/g );
+		if ( backgroundControl && previewImageExist ) {
+			const inheritedBgImage = '0.5' === this.$el.find( '.elementor-control-media__preview' )[ 0 ].style.opacity;
+			if ( inheritedBgImage ) {
+				this.ui.mediaImageSize.css( 'color', '#a0a5aa' );
+			}
+			this.ui.mediaSizeWrapper.css( 'display', 'flex' );
```

Comment:
don't hardcode CSS values in JS, use classes or something

Reply:
There is already a use of this kind of functionality in the same file, and also in other widgets. Sample from the same file : 

`this.ui.mediaImage.css( 'background-image', url ? 'url(' + url + ')' : '' );
			if ( ! value && url ) {
				this.ui.mediaImage.css( 'opacity', 0.5 );
}`

Reply:
background image is different than color / display.
the opacity in the original code is also bad

Reply:
Only check for property existing & use class regarding the text color of the select input.